js/circlepack.js

```diff
@@ -143,29 +155,235 @@ Circlepack.prototype.updateVis = function(){
+  let vis = this;
+  let circle_x = d.x;
+
+  // only show tooltip on hover if none are selected
+  if (!vis.nodeSelected) {
+    // if scaled circle_x position is in the right third of the screen, draw a westward tooltip
+    if (circle_x / vis.width >= 0.67) {
+      vis.tip.hide(d)
+        .direction('w').offset([-5,-12])
+        .show(d)
+    }  else {
+      vis.tip.hide(d)
+        .direction('e').offset([-5,12])
+        .show(d)
+    }
+  }
+  // always show outline on hover
+  vis.highlight(circle)
+}
+
+/*
+ * @param d -- circlepack hierarchy node
+ * @param circle -- d3 circle selection
+ * Shows tooltip when a circle is clicked
+ */
+Circlepack.prototype.showTooltipOnSelect = function(d, circle) {
+  let vis = this;
+
+  // if clicking a node and nothing is selected, select it, highlight it, and show tooltip
+  if (!vis.nodeSelected) {
+    vis.nodeSelected = circle;
+    vis.highlight(circle)
+    vis.tip.hide(d).show(d);
+
+  // if clicking a node that is already selected, unselect it and hide tooltip
+  } else if (circle === vis.nodeSelected) {
+    vis.resetTooltip()
+
+  // if clicking a node and a different node is already selected, unselect it and select this one
+  } else if (vis.nodeSelected && vis.nodeSelected !== circle) {
+    let previous = vis.nodeSelected
+    vis.tip.hide(d).show(d);
+    vis.unHighlight(previous)
+    vis.nodeSelected = circle;
+    vis.highlight(circle)
+  }
+}
+
+Circlepack.prototype.highlight = function(circle) {
+  d3.select(circle)
+    .attr("stroke-width", '1.5px')
+    .attr("stroke", "#000")
+}
+
+Circlepack.prototype.unHighlight = function(circle) {
+  d3.select(circle)
+    .attr("stroke", 'none')
+}
+
+/*
+ * @param d -- circlepack hierarchy node
+ * @return String -- string containing html text that will be displayed on the tooltip
+ */
+Circlepack.prototype.formatTooltip = function(d) {
+  var title_html = ''
+  var children_html = ''
+  var style_tag = ''
+
+  /*temporary for debugging:*/
+  var diff = `<div class="tooltip-diff">${d.data.diff}</div>`
+
+  // if date is present parse string as a Date and format it
+  var date_label = d.data.date ? formatDate(parseDateTime(d.data.date)) : 'Date unknown'
+  var date_html = `<div class="tooltip-date">${date_label}</div>`
+  var title_link = d.data.link
+
+  // if current node you're hovering over is a dataset, construct dataset title only
+  if (isDataset(d)) {
+    title_html = `<div class="tooltip-dataset tooltip-title"><a href="${title_link}" target="_blank">${d.data.name}</a></div>`
+
+  // else if it's a dataverse, construct a dataverse title and add children to the description section
+  } else if (isDataverse(d)) {
+    title_html = `<div class="tooltip-dataverse tooltip-title"><a href="${title_link}" target="_blank">${d.data.name}</a></div>`
+    children_html = children_html.concat(`<ul>`)
+
+    // if dataverse has more than 5 children show link to all datasets
+    if (d.children.length > 5) {
```

Comment:
Update the logic in this if block here to loop over the first 5 nested datasets/dataverses before you add on the link to "All datasets"